core/node/consistency_checker/src/lib.rs

```diff
@@ -829,3 +806,34 @@ async fn wait_for_l1_batch_with_metadata(
             .ok();
     }
 }
+
+// Get settlement layer based on ETH tx, all eth txs should be presented on settlement layer. what is the best place for this function?
+pub async fn get_settlement_mode(
+    db_pool: ConnectionPool<Core>,
+    eth_client: Box<DynClient<L1>>,
+    sl_client: Option<Box<DynClient<L2>>>,
+) -> anyhow::Result<SettlementMode> {
+    let tx_hash = db_pool
+        .connection()
+        .await?
+        .eth_sender_dal()
+        .get_last_eth_tx()
+        .await?;
+
+    if let Some(tx_hash) = tx_hash {
+        let tx_status = eth_client.get_tx_status(tx_hash).await?;
+        if tx_status.is_some() {
+            return Ok(SettlementMode::SettlesToL1);
+        } else {
+            // The tx has not been found on l1, we have to check on l2
```

Comment:
nit: while techinically gateway is L2, I'd prefer to not call it this way and say "gateway" explicitly, otherwise it will be confusing to distinguish between and normal chains that we also call L2s